trending_data/src/main/webapp/unique_window_handler.js

```diff
@@ -75,15 +67,24 @@ export class UniqueWindowHandler {
     /** YouTube button */
     const ytBttn = document.createElement('button');
     ytBttn.textContent = 'YouTube';
-    ytBttn.className = 'btn btn-danger';
+    /** When open window, youtube data will be shown hence
+    youtube button has to be selected */
+    ytBttn.className = 'btn btn-default yt yt-selected';
     bttnDiv.appendChild(ytBttn);
 
     /** Twitter button */
     const twitterBttn = document.createElement('button');
     twitterBttn.textContent = 'Twitter';
-    twitterBttn.className = 'btn btn-info';
+    twitterBttn.className = 'btn btn-default twitter';
     bttnDiv.appendChild(twitterBttn);
 
+    if (this.showing == 'twitter') {
+      /** if showing Twitter make Twitter
+      button selected and unselect YouTube */
+      twitterBttn.className += ' twitter-selected';
+      ytBttn.className = 'btn btn-default yt';
+    }
+
```

Comment:
Shouldn't there be a one like this for yt?

Reply:
Each time any button is clicked the data window is reinitialized because the current content needs to be cleared. That includes recreating the buttons (I suppose this could be easily changed to keep the button container and only clear the actual data). But since buttons are recreated as well, yt button will always be selected by default unless the twitter button has been pressed - in which case all we have to do is unselect the yt button.

Reply:
This was changed so that buttons are not reinitialized every time one of them is clicked. Instead when user clicks on marker, the button container is added to the popup and when a button is clicked, only the data is cleared and the existing buttons selected/unselected